ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -77,6 +77,36 @@ abstract class CoreVerifier(protected val node: Node) {
             }
         }
 
+        /**
+         * Checks the values of NotBefore and NotOnOrAfter attributes and verifies
+         * that the value of NotBefore is less than the value for NotOnOrAfter.
+         */
+        internal fun validateTimeWindow(node: Node, samlCode: SAMLCoreRefMessage) {
+            val notBefore = node.attributeNode("NotBefore")?.apply {
+                CommonDataTypeVerifier.verifyDateTimeValues(this)
+            }
+
+            val notOnOrAfter = node.attributeNode("NotOnOrAfter")?.apply {
+                CommonDataTypeVerifier.verifyDateTimeValues(this)
+            }
+
+            if (notBefore == null || notOnOrAfter == null) return
+
+            val notBeforeValue = Instant.parse(notBefore.textContent)
+            val notOnOrAfterValue = Instant.parse(notOnOrAfter.textContent)
+            if (notBeforeValue == notOnOrAfterValue || notBeforeValue.isAfter(notOnOrAfterValue))
+                throw SAMLComplianceException.create(samlCode,
+                        message = "NotBefore element with value $notBeforeValue is not less " +
+                                "than NotOnOrAfter element with value $notOnOrAfterValue.",
+                        node = node)
+        }
+
+        private fun retrieveCurrentEncryptedElements(responseDom: Node): List<Node> {
```

Comment:
❓ Is there a reason we are moving this into the companion?

Reply:
It has always been in the companion object

Reply:
I think it was originally an instance method.

Reply:
Nope pretty sure it's always been in the companion object. https://github.com/connexta/saml-conformance/blob/master/ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt#L111